smart-contracts/scripts/deployments/lock.js

```diff
@@ -0,0 +1,99 @@
+const { ethers } = require('hardhat')
+const contracts = require('@unlock-protocol/contracts')
+const createLockHash = require('../../test/helpers/createLockCalldata')
+
+const toBigNumber = (mayBN) =>
+  ethers.BigNumber.isBigNumber(mayBN) ? mayBN : ethers.BigNumber.from(mayBN)
+
+async function main({ unlockAddress, unlockVersion, serializedLock, salt }) {
+  const [signer] = await ethers.getSigners()
+  // get the right version of Unlock
+  let Unlock
+  if (!unlockVersion) {
+    Unlock = await ethers.getContractFactory('Unlock')
+  } else {
+    const { abi, bytecode } = contracts[`UnlockV${unlockVersion}`]
+    Unlock = await ethers.getContractFactory(abi, bytecode)
+  }
+  const unlock = Unlock.attach(unlockAddress)
+
+  const { expirationDuration, tokenAddress, keyPrice, maxNumberOfKeys, name } =
+    serializedLock
+
+  // eslint-disable-next-line no-console
+  console.log(`LOCK DEPLOY > creating a new lock '${name}'...`)
+
+  let tx
+  if (unlockVersion < 9) {
+    tx = await unlock.createLock(
+      expirationDuration,
+      tokenAddress,
+      keyPrice,
+      maxNumberOfKeys,
+      name,
+      salt
+    )
+  } else {
+    const calldata = await createLockHash({
```

Comment:
Ho nice, does that mean we support v10 ?

Reply:
yes it should support both version. Also there is no v10, we only reached v9 with the upgradeable pattern. PublicLock is at v10 tho